Makefile

```
@@ -42,101 +42,46 @@ check-all-links: clean build link-checker-setup
-
-# Usage: VITESS_DIR=/full/path/to/vitess.io/vitess make vtctldclient-docs
-vtctldclient-docs:
-	go run ./tools/cobradocs/ --vitess-dir "${VITESS_DIR}" --version-pairs "${COBRADOC_VERSION_PAIRS}" vtctldclient
-
-# Usage: VITESS_DIR=/full/path/to/vitess.io/vitess make vtgate-docs
-vtgate-docs:
-	go run ./tools/cobradocs/ --vitess-dir "${VITESS_DIR}" --version-pairs "${COBRADOC_VERSION_PAIRS}" vtgate
-
-# Usage: VITESS_DIR=/full/path/to/vitess.io/vitess make vtgateclienttest-docs
-vtgateclienttest-docs:
-	go run ./tools/cobradocs/ --vitess-dir "${VITESS_DIR}" --version-pairs "${COBRADOC_VERSION_PAIRS}" vtgateclienttest
-
-# Usage: VITESS_DIR=/full/path/to/vitess.io/vitess make vtorc-docs
-vtorc-docs:
-	go run ./tools/cobradocs/ --vitess-dir "${VITESS_DIR}" --version-pairs "${COBRADOC_VERSION_PAIRS}" vtorc
-
-# Usage: VITESS_DIR=/full/path/to/vitess.io/vitess make vttablet-docs
-vttablet-docs:
-	go run ./tools/cobradocs/ --vitess-dir "${VITESS_DIR}" --version-pairs "${COBRADOC_VERSION_PAIRS}" vttablet
-
-# Usage: VITESS_DIR=/full/path/to/vitess.io/vitess make vttestserver-docs
-vttestserver-docs:
-	go run ./tools/cobradocs/ --vitess-dir "${VITESS_DIR}" --version-pairs "${COBRADOC_VERSION_PAIRS}" vttestserver
-
-# Usage: VITESS_DIR=/full/path/to/vitess.io/vitess make vttlstest-docs
-vttlstest-docs:
-	go run ./tools/cobradocs/ --vitess-dir "${VITESS_DIR}" --version-pairs "${COBRADOC_VERSION_PAIRS}" vttlstest
-
-# Usage: VITESS_DIR=/full/path/to/vitess.io/vitess make zk-docs
-zk-docs:
-	go run ./tools/cobradocs/ --vitess-dir "${VITESS_DIR}" --version-pairs "${COBRADOC_VERSION_PAIRS}" zk
-
-# Usage: VITESS_DIR=/full/path/to/vitess.io/vitess make zkctl-docs
-zkctl-docs:
-	go run ./tools/cobradocs/ --vitess-dir "${VITESS_DIR}" --version-pairs "${COBRADOC_VERSION_PAIRS}" zkctl
-
-# Usage: VITESS_DIR=/full/path/to/vitess.io/vitess make zkctld-docs
-zkctld-docs:
-	go run ./tools/cobradocs/ --vitess-dir "${VITESS_DIR}" --version-pairs "${COBRADOC_VERSION_PAIRS}" zkctld
+BINS := mysqlctl mysqlctld vtaclcheck topo2topo vtbackup vtclient vtcombo \
+        vtctld vtctldclient vtgate vtgateclienttest vtorc vttablet vttestserver \
+        vttlstest zk zkctl zkctld
+
+# Pattern rule for building docs for a single binary.
+# Running `make mysqlctl-docs` will trigger this rule, for example.
+# VITESS_DIR should be specified as an environment variable, pointing to the root of the local Vitess repository
+# with most recent code and all release branches fetched, for which the docs are being generated.
+# Example, to make all current release  versions:
+# `make mysqlctl-docs VITESS_DIR=~/go/src/github.com/vitessio/vitess`
+# For a specific version, you can specify COBRADOC_VERSION_PAIRS as an environment variable, Example:
+# `make mysqlctl-docs COBRADOC_VERSION_PAIRS="main:22.0" VITESS_DIR=~/go/src/github.com/vitessio/vitess`
+%-docs:
+	set -x
+	@if echo "$$COBRADOC_VERSION_PAIRS" | grep -qE '20\.0'; then \
+    		if [ "$${GOROOT##*/}" != "go1.21" ]; then \
```

This equality check is pretty aggressive, the end user must have `go1.21`, patch releases such as `go1.21.13` will not work.

---

We should use double bracket here instead of simple brackets, that enables us to use the `bash` extension that support pattern matching, instead of the POSIX `test` command (`[ ]`).

That way we can do this:
```suggestion
	@if echo "$$COBRADOC_VERSION_PAIRS" | grep -qE '20\.0'; then \
    		if [[ "$${GOROOT##*/}" != *go1.21* ]]; then \
```